ivy/stateful/module.py

```diff
@@ -358,7 +360,7 @@ def _call(self, *args, v=None, with_grads=None, **kwargs):
         """
         with_grads = ivy.with_grads(with_grads)
         if not self._built:
-            self.build(*args, **kwargs, from_call=True)
+            self.build(*args, **kwargs, from_call=True, dtype=args[0].dtype)
```

Comment:
`args[0]` is very likely to be an array, but strictly speaking it doesn't need to be. We should use one of the nested functions to find the first array argument. Check out [this function](https://github.com/unifyai/ivy/blob/523d745f3ea2492ca7714e1ff654428f4ff640ed/ivy/func_wrapper.py#L163).

Reply:
done